### ISSUE TITLE
org.eclipse.osgi.internal.url.MultiplexingFactory is printing a stack trace unnecessarily

### ISSUE THREAD

**Issue by catshow** [eclipse-equinox/equinox#94]
Please remove the printstacktrace that was added in this commit.
line 100:
```java
		} catch (Throwable t) {
 			t.printStackTrace();
 			// ingore as if there is no Unsafe
 		}
```
https://github.com/eclipse-equinox/equinox/commit/77fda35b312c6da5c6ce1530829804f7512bae7c

java.lang.ClassNotFoundException: sun.misc.Unsafe
	at com.apple.transporter.launcher.loader.OSGIBootstrapClassLoader.loadClassData(OSGIBootstrapClassLoader.java:288)
	at com.apple.transporter.launcher.loader.OSGIBootstrapClassLoader.findClass(OSGIBootstrapClassLoader.java:225)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at OSGi-Bootstrapper//com.apple.transporter.bootstrap.TransporterClassLoader.loadClass(TransporterClassLoader.java:89)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:340)
	at org.eclipse.osgi.internal.url.MultiplexingFactory.<clinit>(MultiplexingFactory.java:61)

**Comment by akurtakov** [comment]
@tjwatson wdyt? the comment makes it look like the print is not needed.

**Comment by tjwatson** [comment]
I would like to know what JVM and version this happens on.  When this does happen the framework is not in a great state for cleaning up on shutdown.  In reality we should be logging this as a warning.

**Comment by catshow** [comment]
This occurred with openjdk-17.0.3, using custom osgi.framework initialization and a reduced runtime produced by jlink.  

It does not happen with the org.eclipse.osgi-3.16.200.v20210226-1447.jar, which I reverted to using instead of the current version.

Is there documentation that states the sun.misc.Unsafe is required to use this library, because it seems there is a fallback mechanism in place to handle this scenario.  Therefore I question the need to log it period.

BTW my use case for this is a command line tool, which I would prefer to not generate random stack traces in the output.

Thanks for your consideration.

**Comment by tjwatson** [comment]
The issue is for things like `java.net.URL.setURLStreamHandlerFactory(URLStreamHandlerFactory)` that are required to be used by the framework in order to provide the necessary URL handler support.  We can certainly call that on startup, but the issue is how do we unset that when the framework is shutdown to cleanup.  In many scenarios that may not be a concern at all because the JVM often is on its way to shutting down when the framework is stopped.  But there are other cases where framework instances come and go within the lifetime of the JVM instance.  It is in these cases where we need to be able to unset.  Unfortunately that requires some "deep reflection" that the JVM doesn't allow for in Java 9+.  The use of Unsafe is a work around that restriction.

There are other options, like forcing the JVM arguments to launch the framework to open up the jdk module for reflection.  There is an OSGi issue open related to this: https://github.com/osgi/osgi/issues/226 where discussion is had about developing a common library that can be used to manage a singleton handler instance, but that has gotten little attention lately.

TLDR; I agree, we should remove this stack trace from standard output.  I still think we should log a warning though.  That will only show up in the eclipse log.  It shouldn't be externally visible on the standard out/error.  I think that is good enough.
